packages/devtools_app/lib/src/screens/vm_developer/vm_function_display.dart

```diff
@@ -0,0 +1,129 @@
+// Copyright 2022 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:collection/collection.dart';
+import 'package:flutter/material.dart';
+import 'package:vm_service/vm_service.dart';
+
+import '../../primitives/utils.dart';
+import '../../shared/common_widgets.dart';
+import '../../shared/table.dart';
+import '../../shared/theme.dart';
+import 'vm_developer_common_widgets.dart';
+import 'vm_object_model.dart';
+import 'vm_service_private_extensions.dart';
+
+// TODO(mtaylee): Finish implementation of [ICDataArrayWidget] and add it to
+// the [VmFuncDisplay].
+
+/// A widget for the object inspector historyViewport displaying information
+/// related to 'Func' (function) objects in the Dart VM.
+class VmFuncDisplay extends StatelessWidget {
+  const VmFuncDisplay({
+    required this.function,
+  });
+
+  final FuncObject function;
+
+  @override
+  Widget build(BuildContext context) => VmObjectDisplayBasicLayout(
+        object: function,
+        generalDataRows: vmObjectGeneralDataRows(function),
+        sideCardDataRows: _functionDetailRows(function),
+        sideCardTitle: 'Function Details',
+      );
+}
+
+/// Returns a list of key-value pairs (map entries)
+/// containing detailed information of a VM Func object [function].
+List<MapEntry<String, Widget Function(BuildContext)>> _functionDetailRows(
+  FuncObject function,
+) {
+  String? kindDescription(String? kindValue) {
+    if (kindValue == null) return null;
+
+    final funcKind = FunctionKind.values
+        .firstWhereOrNull((element) => element.kind() == kindValue);
+
+    if (funcKind == null) {
+      return 'Unrecognized function kind: $kindValue';
+    }
+
+    final camelCase = RegExp(r'(?<=[a-z])[A-Z]');
+
+    final kind = StringBuffer();
+
+    if (function.obj.isStatic == true) {
+      kind.write('static ');
+    }
+    if (function.obj.isConst == true) {
+      kind.write('const ');
+    }
+
+    kind.write(
+      kindValue
+          .replaceAllMapped(
+            camelCase,
+            (Match m) => ' ${m.group(0)!}',
+          )
+          .toLowerCase(),
+    );
+
+    return kind.toString();
+  }
+
+  String? boolYesOrNo(bool? condition) {
+    if (condition == true) return 'Yes';
+    if (condition == false) return 'No';
+    return null;
```

Comment:
Does it make sense to return null here? We should probably only be populating entries in the list below if their values are non-null.